<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1113</width>
    <height>599</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_5">
    <item>
     <widget class="QSplitter" name="splitter_3">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QSplitter" name="splitter_2">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <widget class="QWidget" name="layoutWidget">
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QPushButton" name="update_devices_list_button">
             <property name="text">
              <string>Update</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="device_detect_button">
             <property name="text">
              <string>Detect</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QTreeWidget" name="devices_list">
           <property name="mouseTracking">
            <bool>false</bool>
           </property>
           <property name="contextMenuPolicy">
            <enum>Qt::CustomContextMenu</enum>
           </property>
           <property name="sortingEnabled">
            <bool>true</bool>
           </property>
           <attribute name="headerCascadingSectionResizes">
            <bool>false</bool>
           </attribute>
           <column>
            <property name="text">
             <string>Device Name</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Protocol</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Name</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Current Test</string>
            </property>
           </column>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="verticalLayoutWidget">
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QRadioButton" name="use_human_readable_encoding">
             <property name="text">
              <string>Human Readable</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="use_compiler_readable_encoding">
             <property name="text">
              <string>Compiler Readable</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QTabWidget" name="console_tabs">
           <property name="contextMenuPolicy">
            <enum>Qt::CustomContextMenu</enum>
           </property>
           <property name="currentIndex">
            <number>0</number>
           </property>
           <property name="tabsClosable">
            <bool>true</bool>
           </property>
           <property name="movable">
            <bool>true</bool>
           </property>
           <widget class="QWidget" name="ConsoleTab">
            <attribute name="title">
             <string>Console</string>
            </attribute>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QPlainTextEdit" name="console_edit">
               <property name="readOnly">
                <bool>true</bool>
               </property>
               <property name="maximumBlockCount">
                <number>1000</number>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QSplitter" name="splitter">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <widget class="QWidget" name="layoutWidget">
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QTreeWidget" name="tests_list">
           <property name="contextMenuPolicy">
            <enum>Qt::CustomContextMenu</enum>
           </property>
           <property name="selectionMode">
            <enum>QAbstractItemView::ExtendedSelection</enum>
           </property>
           <property name="sortingEnabled">
            <bool>true</bool>
           </property>
           <column>
            <property name="text">
             <string>Test Name</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Device Requirements</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Connected Devices</string>
            </property>
           </column>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QPushButton" name="run_test_script_button">
             <property name="text">
              <string>Run Selected Tests</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="close_finished_tests_button">
             <property name="text">
              <string>Close Finished Tests</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QTabWidget" name="test_tabs">
        <property name="contextMenuPolicy">
         <enum>Qt::CustomContextMenu</enum>
        </property>
        <property name="tabsClosable">
         <bool>true</bool>
        </property>
        <property name="movable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="tab">
         <attribute name="title">
          <string>Test Information</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QPlainTextEdit" name="test_tab_placeholder">
            <property name="readOnly">
             <bool>true</bool>
            </property>
            <property name="placeholderText">
             <string>Select Test above</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QTextEdit" name="textEdit">
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'MS Shell Dlg 2'; font-size:8.25pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:8pt;&quot;&gt;	&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;table border=&quot;0&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px;&quot; cellspacing=&quot;2&quot; cellpadding=&quot;0&quot;&gt;
&lt;tr&gt;
&lt;td&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt; hallo11			&lt;/p&gt;&lt;/td&gt;
&lt;td&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt; hallo12				&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;
&lt;tr&gt;
&lt;td&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt; hallo21			&lt;/p&gt;&lt;/td&gt;
&lt;td&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt; hallo22		&lt;/p&gt;&lt;/td&gt;&lt;/tr&gt;&lt;/table&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1113</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>Settings</string>
    </property>
    <addaction name="actionPaths"/>
    <addaction name="actionHotkey"/>
   </widget>
   <addaction name="menuSettings"/>
  </widget>
  <action name="actionPaths">
   <property name="text">
    <string>Paths ...</string>
   </property>
  </action>
  <action name="actionHotkey">
   <property name="text">
    <string>Hotkeys...</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>devices_list</sender>
   <signal>itemExpanded(QTreeWidgetItem*)</signal>
   <receiver>MainWindow</receiver>
   <slot>align_columns()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>11</x>
     <y>113</y>
    </hint>
    <hint type="destinationlabel">
     <x>3</x>
     <y>112</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>devices_list</sender>
   <signal>itemCollapsed(QTreeWidgetItem*)</signal>
   <receiver>MainWindow</receiver>
   <slot>align_columns()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>30</x>
     <y>149</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>147</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>devices_list</sender>
   <signal>itemChanged(QTreeWidgetItem*,int)</signal>
   <receiver>MainWindow</receiver>
   <slot>align_columns()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>18</x>
     <y>178</y>
    </hint>
    <hint type="destinationlabel">
     <x>7</x>
     <y>178</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>devices_list</sender>
   <signal>itemEntered(QTreeWidgetItem*,int)</signal>
   <receiver>MainWindow</receiver>
   <slot>align_columns()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>22</x>
     <y>216</y>
    </hint>
    <hint type="destinationlabel">
     <x>5</x>
     <y>215</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tests_list</sender>
   <signal>itemChanged(QTreeWidgetItem*,int)</signal>
   <receiver>MainWindow</receiver>
   <slot>align_columns()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>657</x>
     <y>184</y>
    </hint>
    <hint type="destinationlabel">
     <x>750</x>
     <y>181</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tests_list</sender>
   <signal>itemEntered(QTreeWidgetItem*,int)</signal>
   <receiver>MainWindow</receiver>
   <slot>align_columns()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>723</x>
     <y>217</y>
    </hint>
    <hint type="destinationlabel">
     <x>752</x>
     <y>216</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tests_list</sender>
   <signal>itemExpanded(QTreeWidgetItem*)</signal>
   <receiver>MainWindow</receiver>
   <slot>align_columns()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>724</x>
     <y>245</y>
    </hint>
    <hint type="destinationlabel">
     <x>749</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tests_list</sender>
   <signal>itemCollapsed(QTreeWidgetItem*)</signal>
   <receiver>MainWindow</receiver>
   <slot>align_columns()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>726</x>
     <y>279</y>
    </hint>
    <hint type="destinationlabel">
     <x>749</x>
     <y>278</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>align_columns()</slot>
 </slots>
</ui>
